x/community/keeper/incentives.go

```diff
@@ -0,0 +1,58 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// ShouldStartDisableInflationUpgrade returns true if the disable inflation upgrade should be started
+func (k Keeper) ShouldStartDisableInflationUpgrade(ctx sdk.Context) bool {
+	params, found := k.GetParams(ctx)
+	if !found {
+		return false
+	}
+
+	// skip if we have already upgraded - previousBlockTime is first set on upgrade
+	_, found = k.GetPreviousBlockTime(ctx)
+	if found {
+		return false
+	}
+
+	blockTime := ctx.BlockTime()
+	upgradeTime := params.UpgradeTimeDisableInflation
+
+	// a vanilla kava chain should disable inflation on the first block if `upgradeTime` is not set.
+	// thus, we don't upgrade if `upgradeTime` is set and `blockTime` is before `upgradeTime`.
+	if !upgradeTime.IsZero() && blockTime.Before(upgradeTime) {
+		return false
+	}
+
+	return true
```

Comment:
this logic may be more clear if this whole block were changed to
```suggestion
	// a vanilla kava chain should disable inflation on the first block if `upgradeTime` is not set.
	if upgradeTime.IsZero() {
	  return true
	}
	
	return !blockTime.Before(upgradeTime)
```

Reply:
yep its much cleaner this way. thx, will update